src/MongoDB/Monitoring/CommandStartedEvent.c

```diff
@@ -31,14 +31,15 @@ zend_class_entry* php_phongo_commandstartedevent_ce;
 PHP_METHOD(CommandStartedEvent, getCommand)
 {
 	php_phongo_commandstartedevent_t* intern;
-	php_phongo_bson_state             state = PHONGO_BSON_STATE_INITIALIZER;
+	php_phongo_bson_state             state;
 
 	intern = Z_COMMANDSTARTEDEVENT_OBJ_P(getThis());
 
 	if (zend_parse_parameters_none() == FAILURE) {
 		return;
 	}
 
+	PHONGO_BSON_INIT_STATE(state);
```

Comment:
This is another case where I'd suggest moving this immediately after the declaration. Even though it's clearly not needed for the failure case, the overhead isn't much and it minimizes the risk of misuse by later commits.

Reply:
In this case I put the initialisation in this line to avoid the `memcpy` call if we're exiting early due to a parameter parsing failure. I've done this in a few more places.

Reply:
Correction: `memset` :)

Aye, I'm fine with the early `memset` call, since the early returns are all for error cases representing an unlikely code path.

Reply:
Whoops ;)

Fair enough, I'll move the initialisations up to be right after the declaration.